test/refunder.ts

```diff
@@ -237,36 +239,65 @@ describe("Refunder", function() {
 			await res.wait();
 		});
 
-		it('Successfully refund whitelisted contract function', async () => {
+		it.only('Successfully refund whitelisted contract function', async () => {
 
 			const funcIdAsBytes = generateFuncIdAsBytes('setGreeting(string)');
 			
-			let res = await refunder.whitelistRefundable(greeter.address, funcIdAsBytes, true);
+			let res = await refunder.updateRefundable(greeter.address, funcIdAsBytes, true);
 			await res.wait();
 
-			const balanceBefore = await ethers.provider.getBalance(addr1.address);
+			const testUser = addr1;
+			const balanceBefore = await ethers.provider.getBalance(testUser.address);
 			
 			const text = 'Hello, Tester!';
 			const hexString = strToHex(text);
 			const args = ethers.utils.arrayify(hexString);
 			
-			res = await refunder.connect(addr1).relayAndRefund(greeter.address, funcIdAsBytes, args, {
+			res = await refunder.connect(testUser).relayAndRefund(greeter.address, funcIdAsBytes, args, {
 				gasLimit: 200000
 			});
 
 			let txReceipt = await res.wait();
+			let txCost = res.gasPrice.mul(txReceipt.gasUsed)
+			
+			const balanceAfter = await ethers.provider.getBalance(testUser.address);
+			expect(balanceAfter.lt(balanceBefore), 'Sender was over refunded');
```

Comment:
This should throw if `balanceAfter` is > compared to `balanceBefore`, but it does not. We should see what the problem is